src/linalg.jl

```diff
@@ -1,15 +1,3 @@
-# declare struct 
-if VERSION >= v"1.6"
-    # this union alias is added in https://github.com/JuliaLang/julia/pull/39557
-    using SparseArrays: DenseMatrixUnion, AdjOrTransDenseMatrix, DenseInputVector, DenseInputVecOrMat 
```

Comment:
From https://github.com/invenia/BlueStyle
> Import modules with using, with one module per line and at the top of the file when possible.

For this reason, I've moved this block to `src/NiSparseArrays.jl` because `src/chainrules.jl` needs it.